atc/api/info_test.go

```diff
@@ -183,11 +183,12 @@ var _ = Describe("Pipelines API", func() {
 
 				credServer = ghttp.NewServer()
 				vaultManager := &vault.VaultManager{
-					URL:        credServer.URL(),
-					Namespace:  "testnamespace",
-					PathPrefix: "testpath",
-					TLS:        tls,
-					Auth:       authConfig,
+					URL:             credServer.URL(),
+					Namespace:       "testnamespace",
+					PathPrefix:      "testpath",
+					LookupTemplates: []string{"/{{.Team}}/{{.Pipeline}}/{{.Secret}}", "/{{.Team}}/{{.Secret}}"},
```

Comment:
Side note, not a blocker: I'm iffy on using Go template syntax instead of `((vars))` for lookup paths, but I'm OK with it for now as it's what all the other credential managers use and we don't have a great alternative yet.

More detailed thoughts:

My main concern is whether users will be confused by seeing both `{{.Foo}}` and `((foo))` in the same pipeline config, assuming they're using `var_sources`. I think it would be better to just use the same templating syntax consistently throughout the product.

That being said, I don't think worth switching to `((vars))` for this yet. It would be ambiguous with regular static var interpolation: if the user had paths like `/((team))/((pipeline))/((secret))` but then does `fly set-pipeline -v team=foo` it would unintentionally fill in the `((team))` var.

This same concern came up in #4973 and now I'm considering adding a "local var" syntax, which would fix this problem. It might look something like `/((.team))/((.pipeline))/((.secret))`.

...so tl;dr: I'm OK with just using Go templates for now unless you happen to be interested in implementing this "local vars syntax" work. :sweat_smile:

We can revisit this if and when we transition to using [Prototypes](https://github.com/concourse/rfcs/pull/37) for `var_sources` - I'm working on an RFC for that. At that point I think we can pretty much have free reign on implementing a new Vault prototype for use with `var_sources` and we wouldn't have to worry so much about backwards-compatibility since `var_sources` is still experimental. That would also be a good time to clean up SharedPath and friends.

Reply:
The local var idea is interesting for being able to handle these things in the same mechanism as other vars, versus having a completely separate templating mechanism. I'm also a bit iffy on the stupendous power that `text/template` theoretically has - hopefully people aren't doing anything too adventurous here.

I guess the local-var model would be that it's an implicit var source, but otherwise functionally like any other, and populates a small namespace of team name, etc. They could be used in this structure or anywhere else. I anticipate people wanting `((.build-id))` too, looking at the [resource metadata](https://concourse-ci.org/implementing-resource-types.html#resource-metadata), albeit that can unlock some poor patterns.